test/e2e/route_advertisements.go

```diff
@@ -0,0 +1,225 @@
+		// This test ensures the north-south connectivity is happening through podIP
+		ginkgo.When("tests are run towards the external agnhost echo server", func() {
+			ginkgo.It("routes from external bgp server are imported by nodes in the cluster", func() {
+				externalServerV4CIDR, externalServerV6CIDR := getContainerNetworkCIDRs(bgpExternalNetworkName)
+				framework.Logf("the network cidrs to be imported are v4=%s and v6=%s", externalServerV4CIDR, externalServerV6CIDR)
+				for _, node := range nodes.Items {
+					ipVer := ""
+					cmd := []string{containerRuntime, "exec", node.Name}
+					bgpRouteCommand := strings.Split(fmt.Sprintf("ip%s route show %s", ipVer, externalServerV4CIDR), " ")
+					cmd = append(cmd, bgpRouteCommand...)
+					framework.Logf("Checking for server's route in node %s", node.Name)
+					gomega.Eventually(func() bool {
+						routes, err := runCommand(cmd...)
+						framework.ExpectNoError(err, "failed to get BGP routes from node")
+						framework.Logf("Routes in node %s", routes)
+						return strings.Contains(routes, frrContainerIPv4)
+					}, 30*time.Second).Should(gomega.BeTrue())
+					if isDualStackCluster(nodes) {
+						ipVer = " -6"
+						nodeIPv6LLA, err := GetNodeIPv6LinkLocalAddressForEth0(routerContainerName)
+						gomega.Expect(err).NotTo(gomega.HaveOccurred())
+						cmd := []string{containerRuntime, "exec", node.Name}
+						bgpRouteCommand := strings.Split(fmt.Sprintf("ip%s route show %s", ipVer, externalServerV6CIDR), " ")
+						cmd = append(cmd, bgpRouteCommand...)
+						framework.Logf("Checking for server's route in node %s", node.Name)
+						gomega.Eventually(func() bool {
+							routes, err := runCommand(cmd...)
+							framework.ExpectNoError(err, "failed to get BGP routes from node")
+							framework.Logf("Routes in node %s", routes)
+							return strings.Contains(routes, nodeIPv6LLA)
+						}, 30*time.Second).Should(gomega.BeTrue())
+					}
+				}
+
+			})
+
+			ginkgo.It("routes to the default pod network are be advertised to external frr router", func() {
+				// Get the first element in the advertisements array (assuming you want to check the first one)
+				gomega.Eventually(func() bool {
+					podNetworkValue, err := e2ekubectl.RunKubectl("", "get", "ra", "default", "--template={{index .spec.advertisements 0}}")
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return gomega.Expect(podNetworkValue).To(gomega.Equal("PodNetwork"))
+				}).Should(gomega.BeTrue())
+
+				gomega.Eventually(func() bool {
+					reason, err := e2ekubectl.RunKubectl("", "get", "ra", "default", "-o", "jsonpath={.status.conditions[?(@.type=='Accepted')].reason}")
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return gomega.Expect(reason).To(gomega.Equal("Accepted"))
+				}).Should(gomega.BeTrue())
+			})
+
+			ginkgo.It("all 3 node's podSubnet routes are exported correctly to external FRR router by frr-k8s speakers", func() {
```

Comment:
isnt this also an a test thats tests internal implementation details? wdyt

Reply:
yeah I didn't find a better way to check the frr-k8s actually works as expected.. perhaps you are right and we can delete this too..?